Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -284,6 +284,44 @@ extension NIOSSLCertificate: Equatable {
     }
 }
 
+
+extension NIOSSLCertificate: Hashable {
+    /// This is a silly thunk type that exists just to provide stack-allocation for a SHA256 digest.
+    private struct SHA256Digest {
+        // A SHA256 digest is 256 bits long, or 32 bytes, or 4 UInt64s. We store this
+        // as UInt64s here, but they are always accessed as UInt8s.
+        private var storage: (UInt64, UInt64, UInt64, UInt64)
+
+        init() {
+            self.storage = (0, 0, 0, 0)
+        }
+
+        mutating func withUnsafeMutablePointer<T>(_ body: (UnsafeMutableBufferPointer<UInt8>) throws -> T) rethrows -> T {
+            return try Swift.withUnsafeMutablePointer(to: &self.storage) { (storagePtr: UnsafeMutablePointer<(UInt64, UInt64, UInt64, UInt64)>) -> T in
```

Comment:
Does Swift guarantee that tuples are laid out contiguously and won't potentially be reordered/split in memory then?

Reply:
Yes, tuples are guaranteed to be laid out the same way as C arrays. This is why C arrays bridge into Swift as tuples. :smile: